app/components/zig_codeblocks.py

```diff
@@ -121,198 +81,123 @@ async def replace(
+    if omitted_codeblocks:
+        omission_note = omission_note.format(
+            omitted_codeblocks, " was" if omitted_codeblocks == 1 else "s were"
+        )
+        truncation_size = 2000 - len(omission_note) - 1  # -1 for the ellipsis
+        if attachments:
+            content = content.removesuffix(FILE_HIGHLIGHT_NOTE)
+            truncation_size -= len(FILE_HIGHLIGHT_NOTE)
+            omission_note = f"{FILE_HIGHLIGHT_NOTE}{omission_note}"
+        content = f"{content[:truncation_size]}…{omission_note}"
 
-    for msg_content in msg_contents[1:-1]:
-        msg = await message.channel.send(msg_content)
-        codeblock_linker.link(message, msg)
+    return content
 
-    final_msg = await message.channel.send(
-        msg_contents[-1], view=ZigCodeblockActions(message), files=files
-    )
-    codeblock_linker.link(message, final_msg)
-    await remove_view_after_timeout(final_msg, VIEW_TIMEOUT)
 
+async def codeblock_processor(message: discord.Message) -> ProcessedMessage:
+    attachments = await _collect_attachments(message)
+    zig_codeblocks = [c for c in extract_codeblocks(message.content) if c.lang == "zig"]
 
-async def zig_codeblock_edit_hook(
-    before: discord.Message, after: discord.Message
-) -> None:
-    if before.content == after.content and before.attachments == after.attachments:
-        return
+    if not zig_codeblocks:
+        if not attachments:
+            return ProcessedMessage(item_count=0)
+        return ProcessedMessage(
+            content=FILE_HIGHLIGHT_NOTE, files=attachments, item_count=len(attachments)
+        )
 
-    old_contents, old_files = await _prepare_reply(before)
-    new_contents, new_files = await _prepare_reply(after)
-    if old_contents == new_contents and old_files == new_files:
-        return
+    highlighted_codeblocks = [
+        CodeBlock("ansi", apply_discord_wa(highlight_zig_code(c.body, THEME)))
+        for c in zig_codeblocks
+    ]
+    max_length = 2000 - (len(FILE_HIGHLIGHT_NOTE) if attachments else 0)
+    omitted_codeblocks = 0
+    while len(code := "".join(map(str, highlighted_codeblocks))) > max_length:
+        file = _tallest_codeblock_to_file(highlighted_codeblocks)
+        if len(attachments) < 10:
+            attachments.append(file)
+            continue
 
-    if not (replies := codeblock_linker.get(before)):
-        if not (old_contents or before in frozen_messages):
-            # There was no code before, so treat this as a new message.
-            # Note: No new attachments can appear, their count can only decrease.
-            await check_for_zig_code(after)
-        # The message was removed from the M2C map at some point
-        return
+        if not omitted_codeblocks:
+            if max_length == 2000:
+                # We definitely have attachments at this point
+                max_length -= len(FILE_HIGHLIGHT_NOTE)
+            max_length -= 30  # expected omission note size (conservative)
```

Comment:
If the omission note template is a const this could be something like

```suggestion
            max_length -= len(OMISSION_NOTE) + 5  # expected omission note size (conservative)
```